Change reading

Morisset opened the pull request:
I sugest using np.genfromtxt to read the file. It allows to have files containing text columns.
I check that the examples are still working fine.
We loose the possibility to read complexe type.

I also added the xlog and ylog parameters to allow log plots. It apply np.log10() to the data, NOT uses a log-axis.
Ch.

Morisset commented on enplot/run.py:
```diff
@@ -148,7 +148,7 @@ def main():
                 print("Processing data file " + data_file)
 
-            M, m, n = enplot.base.file_data_read(data_file)
+            M, m, n = enplot.base.file_data_read(data_file, usenp=True)
```
This is an easy way to switch between old and new ways to read the data file.

jrjohansson replied:
I think this is good. Let's make np.getfromtxt the new default reading method, while keeping the old one as optional for now. eventually the old method can be dropped if there are no unexpected regressions.

Morisset commented on enplot/base.py:
```diff
@@ -136,6 +135,12 @@ def file_data_read(datafile, sep=None):
+	M, N = data.shape
+	return data, M, N
+
     #
```
If the merge is accepted, better remove the following.

jrjohansson commented:
Hi Morisset. Thanks a lot for the contributions. I agree that these are good changes. I will just need to test this on some of my own use cases. I will try to do this in the coming days and will get back to this soon. Sorry to keep you waiting!

Morisset commented:
Hi!,
No problem, I understand that some test must be done.
All this started with an exercice for my students, and I found that you already did it ;-) I don't think they will find your code (and if they do, that's a good point too!), and I think that I will present your code as the solution. Some of the questions in the exercice where to be able to make a log plot, and read file using genfromtxt, then I though that I could change it a little. And once done, why not share the result ;-)
Cheers,
Christophe

Morisset commented:
BTW, I just realized that the -xlog and -ylog does not wotk, as it conflict with -x.
I changed to -logx and -logy.

Morisset commented:
Hi Robert,

I just realized that you are "the" Robert I point to in all my scientific python lectures! ;-)
Best regards,
Christophe

jrjohansson commented on enplot/run.py:
```diff
@@ -53,6 +53,12 @@ def main():
+    parser.add_argument("-logx",
+			help=("plot using log of the values in x-axis"),
+			action='store_true', default=False)
+    parser.add_argument("-logy",
```
lets make these `--log-x` and `--log-y` to comply with standard unix argument name scheme for multicharacter arguments.

jrjohansson commented on setup.py:
```diff
@@ -18,7 +18,7 @@
 MAJOR = 1
 MINOR = 0
-MICRO = 1
+MICRO = 2
```
Let's make it 1.1.0 instead of 1.0.2 since this PR adds new features (log transforms)

jrjohansson commented:
Thanks for the PR, and sorry for keeping you waiting for so long. I've added a few minor comments. If you could please revise those things we can merge this one!

jrjohansson commented:
Thanks :-) I'm glad that you find the notebooks and this small tool useful!

Morisset commented:
Hi,
I changed the sort function to use numpy (very faster for big files) and updated the REAME.
I don't know how to merge my branch to the your master... I guess you will need to do it.
Cheers,
Christophe

Morisset commented:
I realized that it would be very interesting to have a -C keyword to allow a column to be used as color code. I added it. Updated the README. Would be nice to add an exmaple, but no time now.

jrjohansson commented:
OK, great. Thanks for the updates. Yes, I'll do the merging here. Thanks again!
